### PR TITLE
Fix: CSV and Raw output, escape quotes

### LINKED ISSUE #3050: [BUG] CSV output doesn't handle entries with quotes but not commas.
**What is the bug?**
CSV output of entries like `"a" b` or `a " b` should, according to RFC 4180, escape them to be `"""a"" b"` and `"a "" b"`, respectively.  See section 2.5, https://www.rfc-editor.org/rfc/rfc4180

**How can one reproduce the bug?**
1. Add values to a test index.
```
#!/usr/bin/env bash
# Assuming OPENSEARCH_SERVER is the server,
# OPENSEARCH_USER is the host, OPENSEARCH_PASSWORD is the password.
# and the user has the rights to create and query this index.
# Add an entry to the index that has two quotes in it, not at the boundaries.
curl -XPUT "https://${OPENSEARCH_SERVER}:9200/csv_error/_doc/1" \
    -H 'Content-Type: application/json' \
    -d'{  "field1": "\"a\" b" }' --user "${OPENSEARCH_USER}":"${OPENSEARCH_PASSWORD}"
# Add an entry that has a quote in it.
curl -XPUT "https://${OPENSEARCH_SERVER}:9200/csv_error/_doc/2" \
    -H 'Content-Type: application/json' \
    -d'{ "field1": "\"a\" b" }' --user "${OPENSEARCH_USER}":"${OPENSEARCH_PASSWORD}"
# Optional, add an entry that has no quotes or commas.
curl -XPUT "https://${OPENSEARCH_SERVER}:9200/csv_error/_doc/3" \
    -H 'Content-Type: application/json' \
    -d'{ "field1": "a b" }' --user "${OPENSEARCH_USER}":"${OPENSEARCH_PASSWORD}"
# Optional, add an entry that has commas.
curl -XPUT "https://${OPENSEARCH_SERVER}:9200/csv_error/_doc/4" \
    -H 'Content-Type: application/json' \
    -d'{ "field1": "a, b" }' --user "${OPENSEARCH_USER}":"${OPENSEARCH_PASSWORD}"
```
2. Query the index to CSV output.
```
#!/usr/bin/env bash
curl -XPOST "https://${OPENSEARCH_SERVER}:9200/_plugins/_sql?format=csv" \
    -H 'Content-Type: application/json' \
    -d'{ "query": "SELECT * FROM csv_error LIMIT 50" }' \
     --user "${OPENSEARCH_USER}":"${OPENSEARCH_PASSWORD}"
```
3. Response data is:
```
"a" b  # expected to be """a"" b"
a " b  # expected to be "a "" b"
a b     # correct
"a, b" # correct
```

**What is the expected behavior?**
The CSV output escapes double quotes properly.

**What is your host/environment?**
 - OS: Ubuntu 20.04
 - Version - 2.17.0.0
 - Plugins - default

**Do you have any additional context?**
As far as I can tell, the problem appears to be in function `quoteIfRequired` in `FlatResponseBase.java`, https://github.com/opensearch-project/sql/blob/c1e623d6d3464056a4b16ae7e51dca218d73a5b6/protocol/src/main/java/org/opensearch/sql/protocol/response/format/FlatResponseBase.java#L85 but it's possible changing it there might affect other output formats.

### PR DESCRIPTION
**Pull request by Michael-S**
Fixes #3050

### Description
[Describe what this change achieves]

### Related Issues
Resolves #[Issue number to be closed when this PR is merged]

### Check List
- [x ] New functionality includes testing.
- [ ] New functionality has been documented. (N/A, bug fix)
 - [ ] New functionality has javadoc added. (N/A, bug fix)
 - [ ] New functionality has a user manual doc added. (N/A, bug fix)
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md). (N/A, bug fix)
- [x ] Commits are signed per the DCO using `--signoff`.
- [x] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/sql/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: protocol/src/main/java/org/opensearch/sql/protocol/response/format/FlatResponseBase.java [hunk 1]
```diff
@@ -84,6 +84,10 @@ protected List<List<String>> formatData(List<List<String>> lines) {
 
   protected String quoteIfRequired(String separator, String cell) {
     final String quote = "\"";
-    return cell.contains(separator) ? quote + cell.replaceAll("\"", "\"\"") + quote : cell;
+    if (cell != null && (cell.contains(separator) || cell.contains(quote))) {
+      return quote + cell.replaceAll(quote, quote + quote) + quote;
```
Comment (dai-chen) [inline protocol/src/main/java/org/opensearch/sql/protocol/response/format/FlatResponseBase.java:88]: No matter cell contains separator or quote, we always replace quote?

Comment (Michael-S) [reply protocol/src/main/java/org/opensearch/sql/protocol/response/format/FlatResponseBase.java:88]: If you read the RFC for the CSV output format, https://www.ietf.org/rfc/rfc4180.txt section 2.5 states:

> Each field may or may not be enclosed in double quotes (however some programs, such as Microsoft Excel, do not use double quotes  at all).  If fields are not enclosed with double quotes, then double quotes may not appear inside the fields.

So my understanding is that we would quote if the field contains separator or quote.

**Review by dai-chen — APPROVED** [review]

**Review by Swiddis — APPROVED** [review]

**Comment by MaxKsyunz** [comment]
@Swiddis @dai-chen this PR is failing code coverage. 

Was there a plan to add them soon after?

**Comment by ykmr1224** [comment]
> @Swiddis @dai-chen this PR is failing code coverage.
> 
> Was there a plan to add them soon after?

Addressed in #3155

**Pull request merged**